Library/Homebrew/cmd/update-bash.sh

```diff
@@ -21,12 +21,17 @@ which_git() {
 }
 
 git_init_if_necessary() {
+  set -e
+  trap '{ rm -rf .git; exit 1; }' EXIT
+
   if [[ ! -d ".git" ]]
   then
     git init -q
     git config --bool core.autocrlf false
     git config remote.origin.url https://github.com/Homebrew/homebrew.git
     git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+    git fetch origin
+    git reset --hard origin/master
```

Comment:
> It would be good to handle that specific edge case rather than running git fetch (which is a slow operation) twice on the first brew update.

I think there is at least one reason that we want to run `git fetch` here. This step can take quite a long time. Running `git fetch` here can give nice output to notify the git is init in the first time and along with a progress bar.